client/data/marketplace/use-site-search-es-query.ts

```diff
@@ -0,0 +1,217 @@
+import phpUnserialize from 'phpunserialize';
+import { UseQueryResult, UseQueryOptions, InfiniteData, useInfiniteQuery } from 'react-query';
+import { useSelector } from 'react-redux';
+import { extractSearchInformation, normalizePluginsList } from 'calypso/lib/plugins/utils';
+import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
+import {
+	DEFAULT_FIRST_PAGE,
+	DEFAULT_PAGE_SIZE,
+	DEFAULT_CATEGORY,
+	BASE_STALE_TIME,
+	SITE_SEARCH_CACHE_KEY,
+	SITE_SEARCH_ENDPOINT,
+} from './constants';
+import { ESHits, ESResponse, Plugin, PluginQueryOptions } from './types';
+import { getPluginsListKey } from './utils';
+
+/**
+ *
+ * @param pluginSlug
+ * @param iconsArray
+ * @returns An object containing a resolution -> icon url map
+ */
+const createIconsObject = ( pluginSlug: string, iconsArray: string ) => {
+	type Icon = {
+		resolution: string;
+		filename: string;
+		location: string;
+		revision: string;
+	};
+	const icons: Record< string, Icon > = phpUnserialize( iconsArray );
+	return Object.values( icons ).reduce(
+		( prev: Record< string, string >, { resolution, filename, location, revision } ) => {
+			prev[
+				resolution
+			] = `https://ps.w.org/${ pluginSlug }/${ location }/${ filename }?rev=${ revision }`;
+			return prev;
+		},
+		{}
+	);
+};
+
+const createAuthorUrl = ( header_author: string, header_author_uri: string ) =>
+	`<a href="${ header_author_uri }">${ header_author }</a>`;
```

Comment:
Nitpicking: I would use camelCase when we are not explicitly using a fiedName.

```suggestion
const createAuthorUrl = ( headerAuthor: string, headerAuthorUri: string ) =>
	`<a href="${ headerAuthorUri }">${ headerAuthor }</a>`;
```